packages/analytics/index.tsx

```diff
@@ -1,2 +1,84 @@
+import { AddToCartInput, CartLineItem } from '@ifixit/cart-sdk';
+import {
+   trackGA4AddToCart,
+   trackGA4RemoveFromCart,
+   trackGA4ViewItem,
+} from './google';
+import {
+   trackPiwikV2AddToCart,
+   trackPiwikV2ProductDetailView,
+   trackPiwikV2RemoveFromCart,
+} from './piwik';
+import {
+   getVariantIdFromEncodedVariantURI,
+   getVariantIdFromVariantURI,
+} from '@ifixit/helpers';
+
 export * from './google';
 export * from './piwik';
+
+export type AnalyticsItem = {
+   item_id: string | null | undefined;
+   item_name: string | null;
+   item_variant: string | null;
+   quantity: number | null | undefined;
+   price: number;
+};
+export type AnalyticsItemsEvent = {
+   items: AnalyticsItem[];
+   value: number;
+   currency: string;
+};
+
+export function trackInAnalyticsViewItem(event: AnalyticsItemsEvent) {
+   trackGA4ViewItem(event);
+   trackPiwikV2ProductDetailView(event.items);
+}
+
+export function trackInAnalyticsAddToCart(event: AnalyticsItemsEvent) {
+   trackGA4AddToCart(event);
+   trackPiwikV2AddToCart(event.items);
+}
+
+export function trackInAnalyticsRemoveFromCart(event: AnalyticsItemsEvent) {
+   trackGA4RemoveFromCart(event);
+   trackPiwikV2RemoveFromCart(event.items);
+}
+
+export function convertAddToCartInputToAnalyticsItemEvent(
+   input: AddToCartInput
+): AnalyticsItemsEvent {
+   if (input.type === 'product') {
+      return {
+         items: convertCartLineItemsToAnalyticsItem([input.product]),
+         value: Number(input.product.price.amount),
+         currency: input.product.price.currencyCode,
+      };
+   } else {
+      return {
+         items: convertCartLineItemsToAnalyticsItem(input.bundle.items),
+         value: input.bundle.items.reduce(
+            (acc, item) => acc + Number(item.price.amount),
+            0
+         ),
+         currency: input.bundle.items[0].price.currencyCode,
+      };
+   }
+}
+
+export function convertCartLineItemsToAnalyticsItem(
+   items: CartLineItem[]
+): AnalyticsItem[] {
+   return items.map((item) => {
+      const shopifyVariantId = item.shopifyVariantId.startsWith('gid://')
+         ? getVariantIdFromVariantURI(item.shopifyVariantId)
+         : getVariantIdFromEncodedVariantURI(item.shopifyVariantId);
```

Comment:
Oof, why can't we be certain which form we're dealing with? Does the iFixit API send us different formats depending on which endpoint we use?

Reply:
I still don't understand which service gives us an encoded variant id vs a decoded one

Reply:
for some reason we encode cross sell item's variant url. I'm not entirely sure why, but I'm going to look through if we can just remove that in a future pull

Reply:
Ok thanks! That's all I wanted to know

Reply:
ok this turned out to be a little bit inaccurate. basically we've been mix and matching encoded vs not encoded.

The basis for this seems to be our ifixit cart api returning the shopify variant id encoded, so all cart items have their variants encoded. basically everything else didn't, except for the mini cart cross sell items.